JHUGenerator/mod_Kinematics.F90

```diff
@@ -5410,6 +5411,20 @@ FUNCTION GetBWPropagator(sHat, scheme)
         !BigGamma = BigGamma*GeV
 
         !print *, dsqrt(dabs(sHat))/GeV, gabarH/GeV, BigGamma/GeV
+    elseif( scheme.eq.4) then !jpsi running width.
+        if ( M_Zprime.ne.-1 ) then !if you're using ZPrime then it will consider the ZPrime mass
+            decayMass = M_Zprime
+        else
+            decayMass = M_Z !if you're doing Z-ZPrime stuff this scheme won't work anyways so the point is moot
+        end if
+        ! PRINT *, decayMass, decayMass**2, sHat, 0.25d0*sHat, M_Reso, 0.25d0*M_Reso**2
+        if(0.25d0*(sHat) < decayMass**2) then !sHat is the mass squared!
+            qqq = 0
+        else
+            qqq = sqrt(0.25d0*(sHat) - decayMass**2)
+        endif
+        qqq0=sqrt(0.25d0*(M_Reso**2) - decayMass**2)
+        GetBWPropagator =  1d0/( (sHat-M_Reso**2)**2 + (M_Reso*Ga_Reso*qqq/qqq0)**2 ) !new style running width
```

Comment:
This formula seems to be valid only when mH>2mZ'. At mH=2mZ' (L5426), there is a singularity, and for mH<2mZ', propagator=NaN (L5424). 

The check on LL5421 is not needed; phase space generation takes care of q2 thresholds - in fact, for a Z' with a nonzero width, this would not be a valid check. From the formulation of the running width calculation, it looks like Z's are assumed to be strictly on-shell, which is not what we generate. Would you please provide the reference so that I can check the physics basis of this approximation (hopefully from a more general form factor) and make a suggestion?